src/components/UserDetailsScreen.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+import * as React from 'react';
+import {View, Text, StyleSheet, Image, Linking} from 'react-native';
+import FullName from './FullName';
+import Phone from './Phone';
+import Email from './Email';
+import Location from './Location';
+import type {RouteProp} from '@react-navigation/native';
+import type {StackNavigationProp} from '@react-navigation/stack';
+import type {RootStackParamList, UserState, User} from '../types.js';
+import {useSelector} from 'react-redux';
+
+const styles = StyleSheet.create({
+  avatarContainer: {
+    alignItems: 'center',
+  },
+  avatar: {
+    width: 128,
+    height: 128,
+    marginTop: 24,
+    marginBottom: 8,
+  },
+  detailsContainer: {
+    margin: 16,
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+  },
+  detailLabel: {
+    flexBasis: '25%',
+    fontSize: 16,
+  },
+  detailInfo: {
+    flexBasis: '75%',
+    fontSize: 16,
+  },
+  locationInfo: {
+    flexBasis: '75%',
+    fontSize: 16,
+    color: 'blue',
+    textDecorationLine: 'underline',
+  },
+});
+
+type UserDetailsProps = {
+  navigation: StackNavigationProp<RootStackParamList, 'UserDetails'>,
+  route: RouteProp<RootStackParamList, 'UserDetails'>,
+};
+
+const openMaps = location => {
+  const city = location.city;
+  const state = location.state;
+  const country = location.country;
+
+  let searchableAddress = `${city ? city : ''},${state ? state : ''},${
```

Comment:
Could we use `join` string ?

Reply:
Unfortunately, that does not work if city/state/country have more than 1 word. For example, "London", "United Kingdom" will be converted to "London+United Kingdom".

Reply:
Thank you, I've updated the code to use join instead. I also uploaded a demo video so that you can have a look..